client/SDL/sdl_freerdp.cpp

```diff
@@ -692,30 +681,28 @@ static int sdl_run(sdlContext* sdl)
 	}
 
 	SDL_Init(SDL_INIT_VIDEO);
-	sdl->initialized->set();
+	sdl->initialized.set();
 
-	while (!freerdp_shall_disconnect_context(&sdl->common.context))
+	while (!freerdp_shall_disconnect_context(sdl->context()))
 	{
 		SDL_Event windowEvent = { 0 };
-		while (!freerdp_shall_disconnect_context(&sdl->common.context) &&
-		       SDL_WaitEvent(&windowEvent))
+		while (!freerdp_shall_disconnect_context(sdl->context()) && SDL_WaitEvent(&windowEvent))
 		{
 #if defined(WITH_DEBUG_SDL_EVENTS)
 			SDL_Log("got event %s [0x%08" PRIx32 "]", sdl_event_type_str(windowEvent.type),
 			        windowEvent.type);
 #endif
-			std::lock_guard<CriticalSection> lock(*sdl->critical);
+			std::lock_guard<CriticalSection> lock(sdl->critical);
```

Comment:
`std::lock_guard` is enough

Reply:
@mfleisz sadly our winodws builder is not up to date enough for that syntax.

Reply:
@akallabeth ah yeah - somehow I had in mind we require C++17 (which VC2003 certainly is not capable of) 😅